### DISCUSSION TITLE
GoogleCloudMonitoring: Add support for numerical comparison operators

### DISCUSSION THREAD

**Discussion by plutino** [grafana/grafana#66697] [Data Sources]
This is a follow up on https://github.com/grafana/grafana/discussions/41822 since it is closed and locked.  Please referring to that discussion for details.

The screenshot @andresmgot provided uses a string type label, and regex match works.  For numerical type labels as the original request is about, regex match does not work and GCP API still returns the same error:

```
Field filter had an invalid value of "metric.type="appengine.googleapis.com/http/server/response_count" resource.label.module_id="default" metric.label.response_code=starts_with("4")": Function restrictions are only allowed for string type.
```

Environment:
Grafana version: 9.3.4

**Comment by AndreMiras** [comment]
I can confirm on both Grafana 10.1 and 10.3
For context my metric comes from log-based metrics and I'm extracting the status code.
I want to make the distinction between OK status [200-299] and the other status codes.
Edit: as a workaround I've created a second field on my GCP log-based metric that converts the status code to string

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82583. Feel free to continue the discussion around this there. Thank you!
